types/three/src/objects/InstancedMesh.d.ts

```diff
@@ -5,8 +5,11 @@ import { Mesh } from './Mesh';
 import { Matrix4 } from './../math/Matrix4';
 import { Color } from './../math/Color';
 
-export class InstancedMesh extends Mesh {
-    constructor(geometry: BufferGeometry, material: Material | Material[], count: number);
+export class InstancedMesh<
+    Geom extends BufferGeometry = BufferGeometry,
+    Mat extends Material | Material[] = Material
```

Comment:
This should be preferable instead imo?

```suggestion
    Mat extends Material | Material[] = Material | Material[]
```

Reply:
I think doing that brings about the same problem I'm trying to avoid:

```ts
let mesh: THREE.Mesh

// more code

particleLight = new THREE.Mesh(new THREE.SphereGeometry(4, 8, 8), new THREE.MeshBasicMaterial({ color: 0xffffff }));
particleLight.material.depthTest = true // this throws a TS error because depthTest does not exist on Material[]
scene.add(particleLight);
```

So while it will pick up if someone uses Material[], I don't think it should be a default.

Reply:
Yeah, but I still think the default type parameter should use its broadest definition, as it's hard to assume that just saying `THREE.Mesh` excludes use of multiple materials. Probably a preference / opinionated.

Reply:
@joshuaellis Couldn't you define `particleLight` as `THREE.Mesh<BufferGeometry, Material>`? I agree with @FMS-Cat that it should default to all possible values.